torch/_torch_docs.py

```diff
@@ -5182,13 +5182,24 @@ def merge_dicts(*dicts):
     >>> torch.quantile(a, q)
     tensor([-0.5446,  0.0700,  0.9214])
 
-.. function:: quantile(input, q, dim=None, keepdim=False, *, out=None) -> Tensor
+.. function:: quantile(input, q, dim=None, interpolation='linear', keepdim=False, *, out=None) -> Tensor
 
-Returns the q-th quantiles of each row of the :attr:`input` tensor along the dimension
-:attr:`dim`, doing a linear interpolation when the q-th quantile lies between two
-data points. By default, :attr:`dim` is ``None`` resulting in the :attr:`input` tensor
+Returns the q-th quantiles of each row of the :attr:`input` tensor
+along the dimension :attr:`dim` based on :attr:`interpolation`.
+When the desired quantile lies between two data points ``i < j``,
+the result is computed based on the :attr:`interpolation` value as described below.
+By default, :attr:`interpolation` is ``linear`` and :attr:`dim` is ``None`` resulting in the :attr:`input` tensor
 being flattened before computation.
 
+When the quantile value lies between two data points ``i < j``,
+the result is computed according to the given :attr:`interpolation` method as follows:
+
+- ``linear``: ``i + (j - i) * fraction``, where ``fraction`` is the fractional part of the index surrounded by ``i`` and ``j``.
+- ``lower``: ``i``.
+- ``higher``: ``j``.
+- ``nearest``: ``i`` or ``j``, whichever is nearest.
+- ``midpoint``: ``(i + j) / 2``.
+
 If :attr:`keepdim` is ``True``, the output dimensions are of the same size as :attr:`input`
```

Comment:
This paragraph can probably be removed.

In the future we should think about how all reductions should explain the behavior of `keepdim`, and not require individual reductions, like quantile, to bother.

Reply:
The phrase "represents the quantiles" below is vague. We should be more precise about the behavior when `q` is a scalar vs. a 1D tensor.